lib/pf/import.pm

```diff
@@ -41,79 +42,141 @@ use pf::util;
 
 =item nodes
 
-Handle bulk importation of nodes. They are automatically registered. Status is sent on STDOUT.
-
-Supported: One MAC address per line.
+Handle bulk importation of nodes. They are automatically registered.
 
 =cut
 
-# TODO support more fields
-# TODO instead of printout to STDOUT directly, we should provide a handle to print to in pfcmd
-#      this would allow callers to determine where the output should go. pfcmd -> STDOUT, SOAP -> Apache
 sub nodes {
-    my ($filename) = @_;
+    my ($file, $data, $user, $allowed_roles) = @_;
```

Comment:
$user does not seem to be used

Reply:
Kept the same signature as when it was in pfappserver as my guess is that this could be useful at some point for other checks so IMO I'd leave it